### PR TITLE
[depends] fix rbpi toolchain usage

### PR DESCRIPTION
**Pull request by wsnipex**
fix libc includes from rbpi toolchain and unify handling with other platforms.

## Motivation and Context

Raspberry builds failing to find iconv in toolchain, see https://github.com/xbmc/xbmc/pull/11665

## How Has This Been Tested?

only test compiled, not runtime tested
Note I didn't test if the changed docs are 100% correct.

### CODE REVIEW

**Comment by wsnipex** [comment]
jenkins build this please

**Comment by popcornmix** [comment]
I've tried applying this PR to master and following README.raspberrypi instructions but build ends with:
Makefile:8: recipe for target 'native/.installed-i686-linux-native' failed
Full output: http://paste.ubuntu.com/25902391/

**Comment by wsnipex** [comment]
> Makefile:53: recipe for target 'google-breakpad-native' failed
It fails in native depends, which I didn't touch. I can't find what exactly is going on in the log. Can you try to build only native/google-breakpad-native?

**Comment by popcornmix** [comment]
It seems the update to Ubuntu 17.10 (and gcc 7.2) is responsible for a few build errors in native depends. These build errors aren't related to this PR and with some hacks to avoid them I can cross compile kodi fine with this PR in place.

So this PR looks good to me.

File: docs/README.raspberrypi [hunk 1]
```diff
@@ -39,24 +39,17 @@ build for the first Raspberry Pi, the commands have to be adapted to use
 
     $ cd xbmc/tools/depends
     $ ./bootstrap
-    $ PATH="$PATH:/opt/gcc-linaro-arm-linux-gnueabihf-raspbian-x64/bin" \
-       ./configure --host=arm-linux-gnueabihf \
+    $ ./configure --host=arm-linux-gnueabihf \
        --prefix=/opt/kodi-bcm/kodi-dbg \
-       --with-toolchain=/usr/local/bcm-gcc/arm-bcm2708hardfp-linux-gnueabi/sysroot \
+       --with-toolchain=/opt/gcc-linaro-arm-linux-gnueabihf-raspbian-x64 \
```
Comment (popcornmix) [inline docs/README.raspberrypi]: I'd suggest changing this to:
``--with-toolchain=/opt/arm-rpi-4.9.3-linux-gnueabihf \``
Assuming you are referring to tools from https://github.com/raspberrypi/tools/tree/master/arm-bcm2708
That is a newer 4.9.3 toolchain (vs the 4.8.3 prerelease one from gcc-linaro-arm-linux-gnueabihf-raspbian-x64) that is used when building the pi kernel and /opt/vc/lib libraries.

File: tools/buildsteps/rbpi/configure-depends [hunk 2]
```diff
@@ -6,7 +6,7 @@ if [ "$(pathChanged $WORKSPACE/tools/depends)" == "1" ]
 then
   cd $WORKSPACE/tools/depends;
 
-  PATH="$PATH:$JENKINS_RBPI_DEVENV/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64/bin" \
   ./configure --with-platform=raspberry-pi --host=arm-linux-gnueabihf --prefix=$XBMC_DEPENDS_ROOT --with-tarballs=$TARBALLS \
-    --with-firmware=$JENKINS_RBPI_DEVENV/firmware --build=i686-linux $DEBUG_SWITCH
+    --with-firmware=$JENKINS_RBPI_DEVENV/firmware --build=i686-linux $DEBUG_SWITCH \
+    --with-toolchain=$JENKINS_RBPI_DEVENV/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64
```
Comment (popcornmix) [inline tools/buildsteps/rbpi/configure-depends]: Again, ``arm-rpi-4.9.3-linux-gnueabihf`` would be my preference

File: tools/depends/README [hunk 3]
```diff
@@ -37,7 +37,7 @@ TVOS:
     ./configure --with-toolchain=/opt/toolchains/my-example-toolchain/  --prefix=/opt/xbmc-deps --host=arm-linux-gnueabi
 
   RASPBERRY-PI:
-    PATH="/opt/rbp-dev/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian/bin:$PATH" ./configure --with-platform=raspberry-pi --host=arm-linux-gnueabihf --prefix=/opt/xbmc-deps --with-tarballs=/opt/xbmc-tarballs --with-toolchain=/opt/rbp-dev/tools/arm-bcm2708/arm-bcm2708hardfp-linux-gnueabi/arm-bcm2708hardfp-linux-gnueabi/sysroot --with-firmware=/opt/rbp-dev/firmware --build=i686-linux
+    ./configure --with-platform=raspberry-pi --host=arm-linux-gnueabihf --prefix=/opt/xbmc-deps --with-tarballs=/opt/xbmc-tarballs --with-toolchain=/opt/rbp-dev/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian --with-firmware=/opt/rbp-dev/firmware --build=i686-linux
```
Comment (popcornmix) [inline tools/depends/README]: And here

**Comment by Rechi** [comment]
OSX-64 test failure is unrelated

**Comment by wsnipex** [comment]
@popcornmix updated. Note that this also changes the toolchain on jenkins to gcc-4.9

**Comment by popcornmix** [comment]
> Note that this also changes the toolchain on jenkins to gcc-4.9

That sounds good to me.

**Pull request merged**